src/training/graph_group_sync.cpp

```diff
@@ -357,6 +371,10 @@ void SyncGraphGroup::update(std::vector<Ptr<data::Batch>> subBatches, size_t num
     comm_->scatterReduceAndResetGrads(); // reduce gradients across all devices and MPI nodes into shards
     comm_->foreach(update);              // per-shard model-update
     comm_->allGatherParams();            // distribute param value shards back
+  
+    // Re-compress the model 
```

Comment:
From just reading the code here, it is not clear that this is a stateful operation that retains and re-adds an error residual. I would add a comment here.

Reply:
an additional comment is added..